### PR TITLE
Acknowledgement

### PR DESCRIPTION
**Pull request by svalasovich**

### CODE REVIEW

File: contract/pkg/cache/ddc_bucket_contract_cache.go [hunk 1]
```diff
@@ -58,58 +62,78 @@ func (d *ddcBucketContractCached) ClusterGet(clusterId uint32) (*bucket.ClusterS
...
 		d.nodeCache.SetDefault(key, value)
 		return value, nil
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
-	return cached.(*bucket.NodeStatus), nil
+	return result.(*bucket.NodeStatus), err
```
Comment (abergasov) [inline contract/pkg/cache/ddc_bucket_contract_cache.go:83]:
```go
resp, ok := result.(*bucket.NodeStatus)
if !ok {
return nil, err
}
return resp, nil
```

File: contract/pkg/cache/ddc_bucket_contract_cache.go [hunk 2]
```diff
@@ -58,58 +62,78 @@ func (d *ddcBucketContractCached) ClusterGet(clusterId uint32) (*bucket.ClusterS
...
 		d.bucketCache.SetDefault(key, value)
 		return value, nil
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
-	return cached.(*bucket.BucketStatus), nil
+	return result.(*bucket.BucketStatus), err
```
Comment (abergasov) [inline contract/pkg/cache/ddc_bucket_contract_cache.go:106]:
```go
resp, ok := result.(*bucket.NodeStatus)
if !ok {
    return nil, err
}
return resp, nil
```

File: contract/pkg/cache/ddc_bucket_contract_cache.go [hunk 3]
```diff
@@ -58,58 +62,78 @@ func (d *ddcBucketContractCached) ClusterGet(clusterId uint32) (*bucket.ClusterS
...
+		d.accountCache.SetDefault(key, value)
 		return value, nil
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
-	return cached.(*bucket.Account), nil
+	return result.(*bucket.Account), nil
```
Comment (abergasov) [inline contract/pkg/cache/ddc_bucket_contract_cache.go:129]:
```go
resp, ok := result.(*bucket.Account)
if !ok {
    return nil, err
}
return resp, nil
```

**Review by abergasov — APPROVED** [review]

File: contract/pkg/cache/ddc_bucket_contract_cache.go (hunk 1 above)
Comment (svalasovich) [reply contract/pkg/cache/ddc_bucket_contract_cache.go]: updated to 
```go
resp, _ := result.(*bucket.NodeStatus)
	return resp, err
```

@abergasov WDYT?

**Pull request merged**